### PR TITLE
Telemetry correction

### PR DESCRIPTION
**Pull request by rcohencyberarmor**

### CODE REVIEW

File: pkg/conthandler/container_main_handler.go [hunk 1]
```diff
@@ -160,10 +176,13 @@ func (ch *ContainerHandler) startRelevancyProcess(contEvent v1.ContainerEventDat
...
-				logger.L().Ctx(context.GetBackgroundContext()).Warning("container monitoring got drop events - we may miss some realtime data", helpers.String("container ID", contEvent.GetContainerID()), helpers.String("container name", contEvent.GetContainerName()), helpers.String("k8s resources", contEvent.GetK8SWorkloadID()), helpers.Error(err))
+				ctx, span := otel.Tracer("").Start(ctx, "dropped events.", trace.WithAttributes(attribute.String("containerID", contEvent.GetContainerID()), attribute.String("container workload", contEvent.GetK8SWorkloadID())))
+				logger.L().Ctx(ctx).Warning("container monitoring got drop events - we may miss some realtime data", helpers.String("container ID", contEvent.GetContainerID()), helpers.String("container name", contEvent.GetContainerName()), helpers.String("k8s resources", contEvent.GetK8SWorkloadID()), helpers.Error(err))
+				span.End()
 			}
 		}
 	}
+	logger.L().Ctx(ctx).Info("stop monitor on container - after monitoring time", helpers.String("container ID", contEvent.GetContainerID()), helpers.String("container name", contEvent.GetContainerName()), helpers.String("k8s resources", contEvent.GetK8SWorkloadID()), helpers.Error(err))
```
Comment (matthyx) [inline pkg/conthandler/container_main_handler.go:185]: we don't attach logs above Warning to spans, please remove .Ctx(ctx) from the line

File: pkg/conthandler/container_main_handler.go [hunk 2]
```diff
@@ -160,10 +176,13 @@ func (ch *ContainerHandler) startRelevancyProcess(contEvent v1.ContainerEventDat
 		err = ch.startTimer(watchedContainer, contEvent.GetContainerID())
 		if err != nil {
 			if errors.Is(err, droppedEventsError) {
-				logger.L().Ctx(context.GetBackgroundContext()).Warning("container monitoring got drop events - we may miss some realtime data", helpers.String("container ID", contEvent.GetContainerID()), helpers.String("container name", contEvent.GetContainerName()), helpers.String("k8s resources", contEvent.GetK8SWorkloadID()), helpers.Error(err))
+				ctx, span := otel.Tracer("").Start(ctx, "dropped events.", trace.WithAttributes(attribute.String("containerID", contEvent.GetContainerID()), attribute.String("container workload", contEvent.GetK8SWorkloadID())))
```
Comment (matthyx) [inline pkg/conthandler/container_main_handler.go:179]: don't create a span just for the error, they should be created when you enter a part of your code that you want to trace all the time (aka. record time spent there and compare with other spans)

**Review by matthyx — CHANGES_REQUESTED** [review]
I think we should spend some time together and I'll show you what we expect from telemetry, as it is not just error handling

**Comment by matthyx** [comment]
@rcohencyberarmor I corrected some things myself, PTAL

File: pkg/conthandler/container_aggregator.go [hunk 3]
```diff
@@ -26,7 +26,7 @@ func (aggregator *Aggregator) collectDataFromContainerAccumulator(errChan chan e
 	for {
 		newEvent := <-aggregator.aggregationDataChan
 		if newEvent.GetEventCMD() == accumulator.DropEventOccurred {
-			// aggregator.StopAggregate()
```
Comment (rcohencyberarmor) [inline pkg/conthandler/container_aggregator.go]: I removed it by purpose, the management decided to continue listen on container when we got dropped events.

File: main.go [hunk 4]
```diff
@@ -14,6 +14,9 @@ import (
 )
 
 func main() {
+	context.SetBackgroundContext()
```
Comment (rcohencyberarmor) [inline main.go:17]: it must be after the config parser, so we can get account ID etc..

**Review by matthyx — APPROVED** [review]

**Review by vladklokun — APPROVED** [review]

**Pull request merged**